src/network.rs

```diff
@@ -1,3 +1,39 @@
 //! Network specific utils
 
+use std::fmt::Debug;
+
+use alloy_primitives::U256;
+use alloy_rpc_types::{Header, Transaction};
+use revm::primitives::{BlockEnv, SpecId, TxEnv};
+
+use crate::mv_memory::MvMemory;
+
+/// A chain ID (u64) associated with relevant utils.
+pub trait PevmChain {
+    /// The error type for [Self::build_mv_memory].
+    type BuildMvMemoryError: Debug + Clone;
+
+    /// The error type for [Self::get_block_spec].
+    type GetBlockSpecError: Debug + Clone;
+
+    /// The error type for [Self::get_gas_price].
+    type GetGasPriceError: Debug + Clone;
+
+    /// Get chain id.
+    fn id(&self) -> u64;
+
+    /// Build [MvMemory]
+    fn build_mv_memory(
```

Comment:
Let's provide a default implementation that returns an empty `MvMemory`. Trait implementors can then focus on chain specs to make an integration first before getting to the optimization stage. Facing concepts like `MvMemory`, lazy addresses, etc. too early would be bad DX. I think OP can just use this default implementation as well.